ci-tooling/lib/projects/factory/gitlab.rb

```diff
@@ -41,33 +41,43 @@ def self.understand?(type)
 
     private
 
-    # FIXME: same as in Neon except component is merged
     def split_entry(entry)
       parts = entry.split('/')
-      name = parts[-1]
-      component = parts[0..-2].join('_') || 'gitlab'
-      [name, component]
+      name = parts.pop
+      component = parts.pop
+      group = parts.join('/')
+      [name, component, group]
     end
 
     def params(str)
-      name, component = split_entry(str)
+      name, component, group = split_entry(str)
       default_params.merge(
         name: name,
         component: component,
-        url_base: self.class.url_base
+        url_base: "#{self.class.url_base}/#{group}"
       )
     end
 
     class << self
       def ls(base)
         @list_cache ||= {}
         return @list_cache[base] if @list_cache.key?(base)
+        base_id = ::Gitlab.group_search(base)[0].id
+        repos = list_repos(base_id).collect { |x| x.split('/', 2)[-1] }
+        @list_cache[base] = repos.freeze
+      end
+
+      def list_repos(group_id)
+        repos = []
         # Gitlab sends over paginated replies, make sure we iterate till
         # no more results are being returned.
-
-        base_id = ::Gitlab.group_search(base)[0].id
-        repos = ::Gitlab.group_projects(base_id).auto_paginate
-        @list_cache[base] = repos.collect(&:path).freeze
+        repos += ::Gitlab.group_projects(group_id)
+                         .auto_paginate
+                         .collect(&:path_with_namespace)
+        ::Gitlab.group_subgroups(group_id).auto_paginate.each do |subgroup|
```

Comment:
```suggestion
        repos += ::Gitlab.group_subgroups(group_id).auto_paginate.collect do |subgroup|
			list_repos(subgroup.id)
        end
```

probably nicer to simply collect